actionpack/lib/action_controller/metal/redirecting.rb

```diff
@@ -50,6 +50,9 @@ module Redirecting
     #   redirect_to post_url(@post), status: 301, flash: { updated_post_id: @post.id }
     #   redirect_to({ action: 'atom' }, alert: "Something serious happened")
     #
+    # Statements after redirect_to in our controller gets executed, so redirect_to don't stops the execution of the function.
+     # To terminate the execution of the function immediately after the redirect_to, use return.
+    #   redirect_to post_url(@post) and return
```

Comment:
@eileencodes the `and` here is needed to get the precedence right - `redirect_to post_url(@post) && return` parses as `redirect_to(post_url(@post) && return)` so the redirect doesn't happen.